platformsh-templates/magento2ce
# Use Magento 2.4

### Description (*)
Upgrade template to Magento 2.4 with Elasticsearch 7.7 and Composer 2

### Fixed Issues (if relevant)

1. platformsh-templates/magento2ce#15

### Manual testing scenarios (*)

1. Create project from template
3. Confirm that searches do not return and error
4. Confirm that admin password is prompted for reset after first login
5. Confirm there are no deployment errors

### Questions or comments
**On Lando Development**

This is 99% Lando compatible. The only issue is that after `lando start` you must `lando restart` before the Elasticsearch is exposed to the app. This is a Lando issue, not a Magento or template issue.

This also means that if you are using Lando for local development, you'll need to manually `lando ssh -c "php deploy.php"` for the Magento setup to run locally.

**PR Commits**
Lots of files are changed in this PR. Sorry, they're not squashed. Here's what to expect:

1. Tons of Magento 2.4 files merged into this PR.
2. https://github.com/platformsh-templates/magento2ce/pull/22 is merged into this PR.
3. A number of commits testing deployments with a [new patch added to the deploy.php ](https://github.com/improper/magento2ce/blob/5dd5bc245c2a0db3c598e9f47f4cc92adc42d0d1/deploy.php#L47-L64)that resolves a Magento `setup:install` bug.

### Contribution checklist (*)
